.github/workflows/python-ci-docs.yml

```diff
@@ -24,7 +24,9 @@ on:
     branches:
       - 'main'
   pull_request:
-
+    paths:
+      - '.github/workflows/python-ci-docs.yml'
+      - 'mkdocs/**'
```

Comment:
this action needs to run on code changes as well. See https://github.com/apache/iceberg-python/pull/1334/files#diff-23e8153e0fd497a9212215bd2067068f3b56fa071770c7ef326db3d3d03cee9bL825 where a mistake in docstring caused `mkdoc --strict` to fail